### PR TITLE
[TT-10900] - Add synchroniser support in tyk-data-plane chart

### PR DESCRIPTION
**Pull request by buraksekili**
## Description

Adds a new feature in tyk-data-plane umbrella chart to allow users to configure Tyk MDCB synchroniser easily. If Tyk Control plane uses Tyk MDCB Synchroniser, worker gateways can be configured through values.yaml file to enable synchroniser.

## Related Issue

https://tyktech.atlassian.net/browse/TT-10900

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: components/tyk-mdcb/templates/deployment-mdcb.yaml [hunk 1]
```diff
@@ -115,6 +115,16 @@ spec:
             - name: TYK_MDCB_ANALYTICSCONFIG_MONGOUSESSL
               value: "{{ default "false" .Values.global.mongo.useSSL }}"
             {{ end }}
+            {{ if .Values.global.mdcbSynchronizer.enabled }}
+            - name: TYK_MDCB_SYNCWORKER_ENABLED
+              value: {{ .Values.global.mdcbSynchronizer.enabled | quote }}
+            - name: TYK_MDCB_SYNCWORKER_HASHKEYS
+              value: {{ .Values.mdcb.hashKeys | quote }}
+            {{ if .Values.global.mdcbSynchronizer.privateCertificateEncodingSecret }}
```
Comment (komalsukhani) [inline components/tyk-mdcb/templates/deployment-mdcb.yaml:123]: This field is not specific to Synchronizer. It need to be set if certificates are used.

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: components/tyk-mdcb/templates/deployment-mdcb.yaml (hunk 1 above)
Comment (buraksekili) [reply components/tyk-mdcb/templates/deployment-mdcb.yaml]: waiting for TLS ticket https://github.com/TykTechnologies/tyk-charts/pull/219 to be merged. then, I'll update according to your suggestion

File: components/tyk-mdcb/templates/deployment-mdcb.yaml [hunk 2]
```diff
@@ -115,6 +115,16 @@ spec:
             - name: TYK_MDCB_ANALYTICSCONFIG_MONGOUSESSL
               value: "{{ default "false" .Values.global.mongo.useSSL }}"
             {{ end }}
+            {{ if .Values.global.mdcbSynchronizer.enabled }}
+            - name: TYK_MDCB_SYNCWORKER_ENABLED
+              value: {{ .Values.global.mdcbSynchronizer.enabled | quote }}
+            - name: TYK_MDCB_SYNCWORKER_HASHKEYS
+              value: {{ .Values.mdcb.hashKeys | quote }}
```
[the remark this answers is not in the document]
Comment (komalsukhani) [reply components/tyk-mdcb/templates/deployment-mdcb.yaml:134]: Should be set using global field.

[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-mdcb/templates/deployment-mdcb.yaml:134]: will update. it is left due to conflicts

File: components/tyk-mdcb/values.yaml [hunk 3]
```diff
@@ -292,11 +288,16 @@ mdcb:
...
 
     # location to pem encoded private key
-    certificatekeyFile: "/etc/certs/tls.key"
+    certificateKeyFile: "/etc/certs/tls.key"
 
     # location to pem encoded certificate
     certificateCertFile: "/etc/certs/tls.crt"
 
+    # Sets the AES256 secret which is used to encode certificate private keys when
+    # they uploaded via certificate storage.
```
Comment (komalsukhani) [inline components/tyk-mdcb/values.yaml:297]:
```suggestion
    # they are uploaded via certificate storage.
```

File: tyk-control-plane/values.yaml [hunk 4]
```diff
@@ -1131,6 +1124,30 @@ tyk-mdcb:
...
+      # location to pem encoded private key
+      certificateKeyFile: "/etc/certs/tls.key"
+
+      # location to pem encoded certificate
+      certificateCertFile: "/etc/certs/tls.crt"
+
+      # Sets the AES256 secret which is used to encode certificate private keys when
+      # they uploaded via certificate storage.
+      # It is used to set TYK_MDCB_SECURITY_PRIVATECERTIFICATEENCODINGSECRET
+      privateCertificateEncodingSecret: ""
```
Comment (komalsukhani) [inline tyk-control-plane/values.yaml:1143]: Since this field value should match with gateway value should we move it to global?

**Review by komalsukhani — CHANGES_REQUESTED** [review]

**Pull request merged**